cpp/ql/src/semmle/code/cpp/models/implementations/Fread.qll

```diff
@@ -13,7 +13,8 @@ class Fread extends AliasFunction, RemoteFlowFunction {
 
   override predicate parameterIsAlwaysReturned(int n) { none() }
 
-  override predicate hasFlowSource(FunctionOutput output) {
-    output.isParameterDeref(0)
+  override predicate hasRemoteFlowSource(FunctionOutput output, string description) {
+    output.isParameterDeref(0) and
+    description = "String read by " + this.getName()
```

Comment:
This description looks good enough to me. If we're not planning to provide any more detail, then it doesn't seem necessary to have a `description` parameter on `hasRemoteFlowSource`. The same information is available in the `Tainted{Parameter,Return}Source` classes, and it should be less work to construct the description there than to repeat it for every `RemoteFlowFunction`.

Reply:
I was expecting that we would provide more detail when it's useful, but `fread` and `gets` are the least-specific sources we'll deal with.